cypress/integration/e2e/management-tab/edit-worker.spec.ts

```diff
@@ -1,54 +1,118 @@
-        cy.get('[data-cy="contract"]').click().get('[data-cy="employment_contract"]').click();
+      it("properly handles employment contract", () => {
+        cy.get('[data-cy="employment_contract"]').click();
+        cy.get('[data-cy="contract-time-dropdown"]').click().get('[data-cy="full"]').click();
+        cy.get('[data-cy="contract-time-dropdown"]').click().get('[data-cy="half"]').click();
         cy.get('[data-cy="contract-time-dropdown"]').click().get('[data-cy="other"]').click();
-        cy.get('[data-cy="input-employ-time-other"]').click().type("{backspace}{backspace}13");
-        cy.get('[data-cy="btn-save-worker"]').click();
-        cy.get('[data-cy="btn-add-worker"]').should("be.visible");
-        cy.contains(newWorker);
-        cy.get('[data-cy="btn-schedule-tab"]').click();
-        cy.contains(newWorker);
+        cy.get('[data-cy="input-employ-time-other"] input')
+          .click()
+          .clear({ force: true })
+          .type("123");
+      });
+      it("properly handles civil contract", () => {
+        cy.get('[data-cy="civil_contract"]').click();
+        cy.get('[data-cy="input-civil-time"]').click();
+        cy.get('[data-cy="input-civil-time"] input').clear({ force: true }).type("123");
       });
     });
   });
 
-  describe("Editing the time", () => {
-    beforeEach(() => {
-      cy.get('[data-cy="contract"]').click();
-    });
+  describe("Handle worker data edition", () => {
+    const testWorker = "Pielęgniarka 2";
+    const testWorkerData = {
+      teamIdx: 0,
+      workerIdx: 1,
+      hoursInfo: {
+        [HoursInfoCells.required]: 160,
+        [HoursInfoCells.actual]: 240,
+        [HoursInfoCells.overtime]: 160,
+      },
+    };
 
-    it("properly handles employment contract", () => {
-      cy.get('[data-cy="employment_contract"]').click();
-      cy.get('[data-cy="contract-time-dropdown"]').click().get('[data-cy="full"]').click();
-      cy.get('[data-cy="contract-time-dropdown"]').click().get('[data-cy="half"]').click();
-      cy.get('[data-cy="contract-time-dropdown"]').click().get('[data-cy="other"]').click();
-      cy.get('[data-cy="input-employ-time-other"] input')
-        .click()
-        .clear({ force: true })
-        .type("123");
-    });
-    it("properly handles civil contract", () => {
-      cy.get('[data-cy="civil_contract"]').click();
-      cy.get('[data-cy="input-civil-time"]').click();
-      cy.get('[data-cy="input-civil-time"] input').clear({ force: true }).type("123");
+    describe("Editing workers time ", () => {
+      beforeEach(() => {
+        cy.get('[data-cy="btn-management-tab"]').click();
+        cy.get(`[data-cy="edit-worker-${testWorker}"]`).click();
+      });
+      afterEach(() => {
+        cy.get('[data-cy="btn-schedule-tab"]').click();
+        cy.checkHoursInfo(testWorkerData);
+      });
+      it("properly handles worker hours edition from 1 to 1/2", () => {
```

Comment:
rozbij blok `it` na 
`context("when changing from 1 to 1/2"...`
`testWorkerData.hoursInfo = ...`
`it("properly handles worker hours edition",...`

Reply:
Ok